base/libgit2/utils.jl

```diff
@@ -43,3 +43,15 @@ function features()
     end
     return res
 end
+
+"""
+    LibGit2.posixpath(path)
+
+Standardise the path string `path` to use POSIX separators.
+"""
+function posixpath end
+if is_windows()
+    posixpath(path) = replace(path,'\\','/')
```

Comment:
I fired up my Windows VM, and this is what it gave:
```
julia> r = LibGit2.init("C:\\Foo")
Base.LibGit2.GitRepo(Ptr{Void} @0x00000000068427b0)

julia> LibGit2.gitdir(r)
"C:/Foo/.git/"

julia> rr = LibGit2.GitRepo("\\\\localhost\\c\$\\Foo")
Base.LibGit2.GitRepo(Ptr{Void} @0x0000000006842480)

julia> LibGit2.gitdir(rr)
"//localhost/c\$/Foo/.git/"
```
So it looks like LibGit2 just replaces all back slashes with forward slashes, regardless of whether it is a valid system path.

There is an open issue on UNC paths (https://github.com/libgit2/libgit2/issues/3053) but they don't mention it there.